src/main/java/com/lmax/disruptor/BatchEventProcessor.java

```diff
@@ -38,40 +38,78 @@
     private ExceptionHandler<? super T> exceptionHandler;
     private final DataProvider<T> dataProvider;
     private final SequenceBarrier sequenceBarrier;
-    private final EventHandler<? super T> eventHandler;
+    private final BaseEventHandler<? super T> eventHandler;
     private final int batchLimitOffset;
     private final Sequence sequence = new Sequence(Sequencer.INITIAL_CURSOR_VALUE);
     private BatchRewindStrategy batchRewindStrategy = new SimpleBatchRewindStrategy();
     private int retriesAttempted = 0;
+    private final boolean rewindable;
+
+    private BatchEventProcessor(
+            final DataProvider<T> dataProvider,
+            final SequenceBarrier sequenceBarrier,
+            final BaseEventHandler<? super T> eventHandler,
+            final int maxBatchSize,
+            final boolean rewindable
+    )
+    {
+        this.dataProvider = dataProvider;
+        this.sequenceBarrier = sequenceBarrier;
+        this.eventHandler = eventHandler;
+
+        if (maxBatchSize < 1)
+        {
+            throw new IllegalArgumentException("maxBatchSize must be greater than 0");
+        }
+        this.batchLimitOffset = maxBatchSize - 1;
+
+        this.rewindable = rewindable;
+    }
 
     /**
      * Construct a {@link EventProcessor} that will automatically track the progress by updating its sequence when
      * the {@link EventHandler#onEvent(Object, long, boolean)} method returns.
      *
+     * <p>This constructor will not support rewinding batches.
```

Comment:
How about 'The created BEP will not support batch rewind". The constructor is just a constructor